From 6dbbac0c8fb76c6f4473711fcbd2d2abc8ed437c Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Thu, 31 May 2012 15:45:02 +0800
Subject: [PATCH 282/635] ENGR00211879 Media Server will hold too many file
 handler in camera open/close

Fixed the Camera HAL for memory leakage

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 mx6/libcamera/CameraHal.cpp         | 6 ++++++
 mx6/libcamera/V4l2CapDeviceBase.cpp | 1 +
 mx6/libcamera/V4l2CsiDevice.cpp     | 1 +
 3 files changed, 8 insertions(+)

diff --git a/mx6/libcamera/CameraHal.cpp b/mx6/libcamera/CameraHal.cpp
index e8c3c4e..6571b4e 100755
--- a/mx6/libcamera/CameraHal.cpp
+++ b/mx6/libcamera/CameraHal.cpp
@@ -93,6 +93,12 @@ namespace android {
         CloseCaptureDevice();
         FreeInterBuf();
         postDestroy();
+        if(mVideoMemory != NULL) {
+            mVideoMemory->release(mVideoMemory);
+        }
+        if(mPreviewMemory != NULL) {
+            mPreviewMemory->release(mPreviewMemory);
+        }
     }
 
     void CameraHal :: release()
diff --git a/mx6/libcamera/V4l2CapDeviceBase.cpp b/mx6/libcamera/V4l2CapDeviceBase.cpp
index 3ff530d..74abc55 100755
--- a/mx6/libcamera/V4l2CapDeviceBase.cpp
+++ b/mx6/libcamera/V4l2CapDeviceBase.cpp
@@ -269,6 +269,7 @@ namespace android{
                         fd = 0;
                     }
                 }
+                closedir(v4l_dir);
             }
             if (fd > 0){
                 mCameraDevice = fd;
diff --git a/mx6/libcamera/V4l2CsiDevice.cpp b/mx6/libcamera/V4l2CsiDevice.cpp
index 9e09367..9a61caf 100755
--- a/mx6/libcamera/V4l2CsiDevice.cpp
+++ b/mx6/libcamera/V4l2CsiDevice.cpp
@@ -109,6 +109,7 @@ namespace android{
                         fd = 0;
                     }
                 }
+                closedir(v4l_dir);
             }
             if (fd > 0){
                 mCameraDevice = fd;
-- 
1.8.0

